### PR TITLE
Fix artifacts dir in bootstrap

### PR DESCRIPTION
**Pull request by krzyzacy**
hopefully?
Also added some logging when artifacts dir is missing

ref #4724

/assign @ixdy @BenTheElder

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -940,7 +943,11 @@ def bootstrap(args):
         logging.info('Upload result and artifacts...')
         logging.info('Gubernator results at %s', gubernator_uri(paths))
         try:
-            finish(gsutil, paths, success, '_artifacts', build, version, repos, call)
+            finish(
+                gsutil, paths, success,
+                '%s/_artifacts' % os.getenv(WORKSPACE_ENV, os.getcwd()),
```
Comment (BenTheElder) [inline jenkins/bootstrap.py]: why not `os.path.join(os.getenv(WORKSPACE_ENV, os.getcwd()), '_artifacts')` ? :-)

Comment (krzyzacy) [reply jenkins/bootstrap.py]: done

**Comment by BenTheElder** [comment]
TODO in a followup: fix any other code that makes assumptions about `/workspace`. https://github.com/kubernetes/test-infra/issues/4724#issuecomment-332019860

**Comment by krzyzacy** [comment]
> TODO in a followup: fix any other code that makes assumptions about /workspace.
Not in bootstrap.py as I can tell, but yes, in a million other misc places.

**Comment by krzyzacy** [comment]
:snowflake:

**Comment by BenTheElder** [comment]
@kargakis [it looks like `k8s.io/test-infra/prow/jenkins/go_default_test` possibly has a data race](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/4725/pull-test-infra-bazel/7955/#prowjenkinsgo_default_test).

**Comment by ixdy** [comment]
/lgtm

thanks for cleaning up after my mess!

**Comment by krzyzacy** [comment]
(hope this won't screw up more things :crossed_fingers: )

**Pull request merged**